extensions/example-extension/renderer.tsx

```diff
@@ -1,11 +1,51 @@
 import { Component, Interface, K8sApi, LensRendererExtension } from "@k8slens/extensions";
-import { ExamplePage, ExamplePageParams, namespaceStore } from "./page";
+import { 
+  ExamplePage, ExamplePageParams, namespaceStore,
+  SimplePage, SimpleParamsPage, SimpleParamsPageParams,
+  NonStringParamsPage, NonStringParamsPageParams
+} from "./page";
 import React from "react";
 import path from "path";
 
 export default class ExampleExtension extends LensRendererExtension {
   clusterPages: Interface.PageRegistration[] = [
     {
+      id: "simple-page",
+      components: {
+        Page: () => {
+          return <SimplePage extension={this}/>;
+        },
+      },
+    },
+    {
+      id: "page-with-simple-params",
+      components: {
+        Page: (props: Interface.PageComponentProps<SimpleParamsPageParams>) => {
+          return <SimpleParamsPage {...props} extension={this}/>;
+        },
+      },
+      params: {
+        // setup basic param "exampleId" with default value "demo"
+        exampleId: "demo",
+        namespace: "default"
+      }
+    },
+    {
+      id: "page-with-non-string-params",
+      components: {
+        Page: (props: Interface.PageComponentProps<NonStringParamsPageParams>) => {
+          return <NonStringParamsPage {...props} extension={this}/>;
+        },
+      },
+      params: {
+        exampleId: "demo",
+        namespaceId: {
+          defaultValue: 7
```

Comment:
Btw, here you cannot specify `defaultValue` as number and expect `props.param` helper to work with that `typeof defaultValue` (since with updating param happening to string conversion for URL).

This is because not defined custom functions `parse/stringify` to work with custom type. 
Probably this could be improved with basic types detection like in `defaultValue` and providing pair to custom `parse/stringify` so it allow to avoid some boilerplate in most cases.

Reply:
Ah, that wasn't clear to me. I thought there are default implementations of `parse`/`stringify` that handle simple types? It seems to work with the value provided at the cluster page menu item registration (42). 

So unless your param is a string you must provide `parse`/`stringify` implementations?